### PR TITLE
Code Quality: Introduced pane generation 2

### LINKED ISSUE #11955: Code Quality: Introduce item template to PaneHolderPage.xaml
### Description

The multipane feature consists of two ModernShellPages and a GridSplitter. The code around this is hard to work with and prevents us from expanding the feature to allow more than two panes. Creating an item template will allow us to create and close panes on demand.

### Concerned code

PaneHolderPage.xaml

### Gains

- easier to maintain
- will allow supporting more than two mains

### Requirements

- use item template instead of hardcoding the right and left pane

### Comments

_No response_

### PR DESCRIPTION
**Pull request by 0x5bfa**
### Summary

To keep changes small, I haven't remove left/right pane concept completely.

Phase 3:
- Support vertical panes
- Remove all left/right concept (prepare for more than 2 panes just in case, TBD)

### Resolved / Related Issues

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Closes #11955

### Steps used to test these changes

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Open Files app
2. Open secondary pane
3. See sizer works correctly
4. Close secondary pane

### CODE REVIEW

**Comment by yaira2** [comment]
Overall these changes are good, the one issue I noticed is that continue where you left off isn't working.

**Comment by mdtauk** [comment]
Not sure if this is relevant here, but you should prepare some commands for switching orientation of the pane split, the commands for closing a pane, and these can be called whenever the UI is decided on and put in place.

**Comment by 0x5bfa** [comment]
@mdtauk Good idea. I keep in mind that.

Technically speaking, what I just have to do for this preparation in this PR is to set Grid.Row property to each every child (shell pages and sizer control). If user switches pane view mode, the shell just has to remove column definitions and add row definitions. So code is still clean:)

**Comment by yaira2** [comment]
@0x5bfa any luck on the issue with continue where you left off?

**Comment by 0x5bfa** [comment]
Took a time but here's the fix, committed.

**Review by yaira2 — DISMISSED** [review]
LGTM

**Comment by yaira2** [comment]
Functionality and code looks good to me

File: src/Files.App/UserControls/TabBar/TabBar.xaml.cs
Comment (hishitetsu) [inline src/Files.App/UserControls/TabBar/TabBar.xaml.cs:1]: Changes to this file doesn't seem to make sense.

**Review by hishitetsu — COMMENTED** [review]
I found the following issue.
1. Change the width of the pane.
2. Close the secondary pane.
3. Reopen the secondary pane.
4. The width of the secondary pane will become 0.

File: src/Files.App/UserControls/TabBar/TabBar.xaml.cs
Comment (0x5bfa) [reply src/Files.App/UserControls/TabBar/TabBar.xaml.cs:1]: Forget to remove, will remove.

Comment (0x5bfa) [reply src/Files.App/UserControls/TabBar/TabBar.xaml.cs:1]: Fixed and removed.

**Review by hishitetsu — APPROVED** [review]
LGTM

**Pull request merged**